### ISSUE TITLE
Blocking shard moves fails if there is a foreign key between partitioned distributed tables (from child to parent)

### ISSUE THREAD

**Issue by onderkalaci** [citusdata/citus#4949]
```SQL
CREATE SCHEMA shard_move_fkeys_indexes;
SET search_path TO shard_move_fkeys_indexes;
SET citus.next_shard_id TO 8970000;
SET citus.next_placement_id TO 8770000;
SET citus.shard_count TO 4;
SET citus.shard_replication_factor TO 1;

CREATE TABLE sensors(
measureid         integer,
eventdatetime     date,
measure_data      jsonb,
PRIMARY KEY (measureid, eventdatetime, measure_data)) 
PARTITION BY RANGE(eventdatetime);

CREATE TABLE sensors_old PARTITION OF sensors FOR VALUES FROM ('2000-01-01') TO ('2020-01-01');
CREATE TABLE sensors_2020_01_01 PARTITION OF sensors FOR VALUES FROM ('2020-01-01') TO ('2020-02-01');
CREATE TABLE sensors_news PARTITION OF sensors FOR VALUES FROM ('2020-05-01') TO ('2025-01-01');

CREATE INDEX index_on_parent ON sensors(measureid);
CREATE INDEX index_on_child ON sensors_2020_01_01(measureid);

SELECT create_distributed_table('sensors', 'measureid');


CREATE TABLE colocated_partitioned_table(
    measureid         integer,
    eventdatetime     date,
    PRIMARY KEY (measureid, eventdatetime))
PARTITION BY RANGE(eventdatetime);

CREATE TABLE colocated_partitioned_table_2020_01_01 PARTITION OF colocated_partitioned_table FOR VALUES FROM ('2020-01-01') TO ('2020-02-01');
SELECT create_distributed_table('colocated_partitioned_table', 'measureid');



-- from child to parent
ALTER TABLE sensors_2020_01_01 ADD CONSTRAINT fkey_from_child_to_parent FOREIGN KEY (measureid,eventdatetime) REFERENCES colocated_partitioned_table(measureid,eventdatetime);

-- load some data
INSERT INTO colocated_partitioned_table SELECT i, '2020-01-05' FROM generate_series(0,1000)i;
INSERT INTO sensors SELECT i, '2020-01-05', '{}' FROM generate_series(0,1000)i;


SELECT citus_move_shard_placement(8970000, 'localhost', 9700, 'localhost', 9701, shard_transfer_mode:='block_writes');
ERROR:  insert or update on table "sensors_2020_01_01_8970008" violates foreign key constraint "fkey_from_child_to_parent_8970008"
DETAIL:  Key (measureid, eventdatetime)=(1, 2020-01-05) is not present in table "colocated_partitioned_table_8970020".
CONTEXT:  while executing command on localhost:9701
Time: 257.287 ms

```

**Comment by SaitTalhaNisanci** [comment]
This seems to fail because we `attach` the partitions after we fill the tables, and we don't fill the parent table hence the data is missing. So we should probably change the ordering of DDL commands, a super quick try, which seems to fail on some other tests but fixes this is:

https://github.com/citusdata/citus/tree/fix/partitionCommandsOrdering

**Comment by metdos** [comment]
@niupre I suppose this is fixed both in the master and release 11.0 branch. Shall we close this issue?

**Comment by onderkalaci** [comment]
fixed by #6077
